ground/src/main/java/com/google/android/ground/persistence/remote/firebase/ModelToProtoExt.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.persistence.remote.firebase
+
+import com.google.android.ground.model.AuditInfo
+import com.google.android.ground.model.geometry.Coordinates
+import com.google.android.ground.model.geometry.Geometry
+import com.google.android.ground.model.geometry.LineString
+import com.google.android.ground.model.geometry.LinearRing
+import com.google.android.ground.model.geometry.MultiPolygon
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.geometry.Polygon
+import com.google.android.ground.model.locationofinterest.LocationOfInterest
+import com.google.android.ground.model.locationofinterest.LoiProperties
+import com.google.android.ground.proto.AuditInfo as AuditInfoProto
+import com.google.android.ground.proto.Coordinates as CoordinatesProto
+import com.google.android.ground.proto.Geometry as GeometryProto
+import com.google.android.ground.proto.LinearRing as LinearRingProto
+import com.google.android.ground.proto.LocationOfInterest as LocationOfInterestProto
+import com.google.android.ground.proto.LocationOfInterest.Property as PropertyProto
+import com.google.android.ground.proto.LocationOfInterest.Source as SourceProto
+import com.google.android.ground.proto.MultiPolygon as MultiPolygonProto
+import com.google.android.ground.proto.Point as PointProto
+import com.google.android.ground.proto.Polygon as PolygonProto
+import com.google.protobuf.Timestamp
+import java.util.Date
+
+fun LocationOfInterest.toProtoBuf(): LocationOfInterestProto =
+  LocationOfInterestProto.newBuilder()
```

Comment:
Please use the Kotlin proto DSL instead of the Java Builder for better readability. Reference guide:

https://protobuf.dev/reference/kotlin/kotlin-generated/